File: implementations/sangria/src/main/scala/service/ProductResearchService.scala
package service

import model._

import scala.concurrent.Future

trait ProductResearchService {
  def productResearch(studyCaseNumber: ID): Future[Option[ProductResearch]]
}

object ProductResearchService {
  private val productResearch1 = ProductResearch(CaseStudy(ID("1234"), Some("Federation Study")))
  private val productResearch2 = ProductResearch(CaseStudy(ID("1235"), Some("Studio Study")))
  private val productResearches: List[ProductResearch] = List(productResearch1, productResearch2)

  val inMemory: ProductResearchService = new ProductResearchService {
    override def productResearch(studyCaseNumber: ID): Future[Option[ProductResearch]] =
      Future.successful(productResearches.find(_.study.caseNumber == studyCaseNumber))
  }
}

File: implementations/nestjs-code-first/src/product-research.service.ts
import { Injectable } from "@nestjs/common";

import { ProductResearch } from "./product-research.model";

@Injectable()
export class ProductResearchService {
  public productResearch: ProductResearch[] = [
    {
      study: {
        caseNumber: "1234",
        description: "Federation Study",
      },
    },
    {
      study: {
        caseNumber: "1235",
        description: "Studio Study",
      },
    },
  ];
}
